docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+* Advanced users may be interested in, and willing to, learn advanced features to build much more interesting charts
+
+
+Based on these, our design goal is to make it **easy to find, use, author, release, and maintain charts**, while still making it possible for advanced users to build sophisticated charts. Furthermore, we have begun the process of **attaching provenance to charts**.
+
+
+We have attempted to build a proposal that does not lose any of the functionality of DM, and which loses only Helm functionality that we believe is disposable. But both tools will need to change substantial portions of their chart-handling logic.
+### Non-Goals
+This proposal doesn’t define how either the client or the server must use this data, though we make some non-normative recommendations (e.g. “this data may be displayed to the user” or “a suitable backend will be able to translate this to a fully qualified URL”).
+
+
+Consequently, this document does not describe the implementation of either client or server sides of this spec. While it defines the pieces necessary for developing and deploying a chart, it does not define a development workflow. Some considerations are listed in [Appendix B](#appendix-b-the-standard-chart-repository).
+
+
+## The Chart Format
+Borrowing the nomenclature from Helm, we define a Chart as a bundle of data and metadata whose contents are sufficient for installing a workload into a Kubernetes cluster running DM.
+
+
+A Chart is composed of the following pieces:
+
+
+1. A human- and machine-readable metadata file called `Chart.yaml`
+2. A directory named `templates/`
+3. An optional `README.md` file
+4. An optional `LICENSE` file
+5. An optional `docs/` directory
+6. An optional `hooks/` directory
+7. An optional `icon.svg.`
+
+
+A chart _may_ have other directories and files, but such files are not defined by or required by this proposal. For the purposes of this proposal, any additional files do not contribute to the core functionality of chart installation.
+
+
+_Note: When a chart file is deployed, a [provenance file](#the-provenance-file) is generated for the chart. That file is not stored inside of the chart, but is considered part of the chart’s packaged format._
+
+
+The chart.yaml file format is described [later in this document](#the-chart-file).
+
+
+The `templates/` directory contains one or more template files, as defined in the [Directory Layout](#directory-layout) section. Templates, along with all of their supporting files, are located therein.
+
+
+An optional `README.md` file may be specified, which contains long-form text information about using this chart. Tools may display this information, if present. The README.md file is in Markdown format, and should contain information about a Chart’s purpose, usage, and development.
+
+
+An optional `LICENSE` file may be present, which specifies license information for this chart and/or the images dependent on it.
```

Comment:
IMO, in the open source world, having a LICENSE file is better than not having one, since if you don't have one, you're left guessing about what's acceptable usage.